File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i = m-1;
        int j = n-1;
        int t = m+n-1;

        while (i>=0 && j>=0){
            if (nums1[i] >= nums2[j]){
                nums1[t]=nums1[i];
                t--;
                i--;
            }else{
                nums1[t]=nums2[j];
                t--;
                j--;
            }
        }
        while (j >= 0){
            nums1[t] = nums2[j];
            t--;
            j--;
        }
    }
}

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        i, j, t = m-1, n-1, m+n-1

        while j > -1:
            if i > -1:
                if nums1[i] > nums2[j]:
                    nums1[t] = nums1[i]
                    i -= 1
                else:
                    nums1[t] = nums2[j]
                    j -= 1
            else:
                nums1[t] = nums2[j]
                j -= 1
            t -= 1

            
        